src/main/java/net/sf/mzmine/modules/peaklistmethods/peakpicking/deconvolution/DeconvolutionTask.java

```diff
@@ -59,6 +61,11 @@
     // User parameters
     private final ParameterSet parameters;
 
+	private RSessionWrapper rSession;

```

Comment:
Well, I know that this isn't marvelous from the design point of view. But, R isn't even called in the features that do not require it. At first, I did as you you say, calling R stuffs only from inside the CentWave.resolvePeak(), but the thing is that resolvePeak() is called many times, and forking Rserve several times is not consistent. It was at the time the JRI engine reference was a singleton instantiated once and for all. Now, the Rsession.open() has to reside outside of the resolvePeak()...

I could have a look again, but, I am not sure about how to do better here.